We read in the spreadsheet as a table, Tab, for the exchange-rate data.  It reads in all of the variables.  We get a warning because the first lines have headers.

In [1]:
import numpy as np
import pandas as pd

Tab = pd.read_excel('China_ExRates.xlsx')
Tab

C:\Users\mcnel\AppData\Local\Temp/ipykernel_10840/3115810909.py:4: FutureWarning: Inferring datetime64[ns] from data containing strings is deprecated and will be removed in a future version. To retain the old behavior explicitly pass Series(data, dtype={value.dtype})
  Tab = pd.read_excel('China_ExRates.xlsx')


Date   USD-CHF   USD-AUD  RMBI-CHF  RMBI-AUD  RMBI-SGD  RMBI-THB  \
0    1999-10-08  0.666533  0.656400   5.51095   5.41135   4.92405   0.20530   
1    1999-10-11  0.667022  0.653600   5.51520   5.42985   4.93570   0.20785   
2    1999-10-12  0.673174  0.654600   5.59920   5.43115   4.93430   0.21010   
3    1999-10-13  0.676361  0.654000   5.63075   5.36605   4.91260   0.20930   
4    1999-10-14  0.677828  0.650500   5.61180   5.36830   4.92845   0.21010   
...         ...       ...       ...       ...       ...       ...       ...   
5192 2019-09-03  1.013346  0.676398   7.26640   4.84930   5.15700   0.23424   
5193 2019-09-04  1.019534  0.679995   7.27570   4.85170   5.15640   0.23354   
5194 2019-09-05  1.013962  0.681180   7.23860   4.86300   5.15520   0.23265   
5195 2019-09-08  1.012146  0.684669   7.19340   4.86850   5.14690   0.23157   
5196 2019-09-09  1.007668  0.686530   7.16790   4.88310   5.15310   0.23203   

      RMBI-MYR  RMBI-RUB  RMBI-CAD  ...  Unnamed: 59  Unnamed: 60  \
0     2.178421  0.320915   5.62495  ...      GBP-Aud     Euro-Aud   
1     2.178211  0.320884   5.61985  ...    -0.003606    -0.005482   
2     2.178211  0.321008   5.61230  ...     0.000831    -0.006734   
3     2.178342  0.322403   5.58495  ...    -0.000435    -0.004019   
4     2.178263  0.321203   5.59075  ...       -0.008    -0.007266   
...        ...       ...       ...  ...          ...          ...   
5192  1.703700  0.107543   5.37450  ...     0.005307     0.006839   
5193  1.703500  0.107946   5.39640  ...    -0.007449    -0.000487   
5194  1.702000  0.108025   5.39610  ...    -0.004297     0.002271   
5195  1.699200  0.108194   5.39850  ...     0.008361     0.006298   
5196  1.704800  0.108695   5.40140  ...    -0.002353     0.000145   

      Unnamed: 61  Unnamed: 62  Unnamed: 63  Unnamed: 64  Unnamed: 65  \
0         KRW-Aud        Dummy       DumDol      THB/AUD      MYR/AUD   
1       -0.002651            0            0    -0.003561    -0.003817   
2       -0.001894            0            0    -0.015835    -0.003064   
3        0.004697            0            0     0.004246       0.0066   
4       -0.006161            0            0    -0.013568     -0.00858   
...           ...          ...          ...          ...          ...   
5192     0.005745            1     0.007641     0.006793      0.00465   
5193    -0.001332            1     0.005304     0.003572     0.001229   
5194    -0.003443            1     0.001742     0.006244     0.001719   
5195       0.0013            1     0.005108    -0.017388      0.00245   
5196     0.001837            1     0.002715     0.023975    -0.001049   

      Unnamed: 66  Unnamed: 67  Unnamed: 68  
0         SGD/AUD      CAN/AUD      RUB/AUD  
1       -0.004637    -0.004888    -0.006488  
2       -0.002372     0.003973    -0.004151  
3        0.008368     0.001926    -0.003571  
4       -0.008368    -0.004828     0.005352  
...           ...          ...          ...  
5192     0.005925     0.007957     0.007475  
5193     0.000958    -0.003998    -0.003039  
5194     0.002761       0.0029     0.001378  
5195     0.002827     0.000488    -0.001086  
5196     0.001194      0.00247    -0.000999  

[5197 rows x 69 columns]

We now look at the size of the Tab. 

In [2]:
Tab.shape

(5197, 69)

We now create the date vector as well as the exchange rate variables.

In [4]:
Tab.iloc[0,0:20]

Date        1999-10-08 00:00:00
USD-CHF                0.666533
USD-AUD                  0.6564
RMBI-CHF                5.51095
RMBI-AUD                5.41135
RMBI-SGD                4.92405
RMBI-THB                 0.2053
RMBI-MYR                2.17842
RMBI-RUB               0.320915
RMBI-CAD                5.62495
YEN-CHF                   72.11
YEN-AUD                    70.7
GBP-CHF                0.403015
GBP-AUD                0.396936
EUR-CHF                0.627156
EUR-AUD                0.617627
KRW-CHF                  819.89
KRW-AUD                  789.46
THB-CHF                 26.8877
MYR-CHF                  2.5534
Name: 0, dtype: object

We import data for the USD, CNY, YEN, GBP, EUR, KRW, THB, SGD,CHF and MYF against CHF, the Swiss Franc, which will be the numeraire currency. Some are flexible rates, some are fixed or quasi-fixed against the dollar.  This will make a big 
difference.  We look at the Tab to see which column is which for which exchange rate.  

In [3]:
date1 = Tab.iloc[1:,0];  USDCHF = Tab.iloc[1:,1]; CNYCHF = Tab.iloc[1:,3]; YENCHF = Tab.iloc[1:,10]; 
GBPCHF = Tab.iloc[1:,12]; EURCHF = Tab.iloc[1:, 14]; KRWCHF = Tab.iloc[1:,16]; THBCHF = Tab.iloc[1:, 18];
SGDCHF = Tab.iloc[1:,20]; MYRCHF = Tab.iloc[1:,19];

We need to transform the variables we imported as Table variables into array variabes, so that we can do numberical analysis.

In [4]:
date2 = np.array(date1); USDCHF1 = np.array(USDCHF); CNYCHF1 = np.array(CNYCHF); YENCHF1 = np.array(YENCHF);
GBPCHF1 = np.array(GBPCHF); EURCHF1 = np.array(EURCHF); KRWCHF1 = np.array(KRWCHF); THBCHF1 = np.array(THBCHF);
SGDCHF1 = np.array(SGDCHF); MYRCHF1 = np.array(MYRCHF)

We now take log differences of the variables.  This transformation expresses each variable as a daily percentage rate of change. Logs remove the dimensionalty program.  We are interested in how much, percentage-wise, the RMB moves against the Swiss Franc when the dollar moves against the Swiss Franc.   

In [5]:
y = np.diff(np.log(CNYCHF1))
x = np.diff(np.log([USDCHF1,YENCHF1,GBPCHF1,EURCHF1,KRWCHF1])).T

We have created a matrix of regressors x, which inclue the US Dollar, Yen, British pound, Euro and Korean Won against the Swiss franc. We also added in a constant term. Use the same Python codes.

In [7]:
import statsmodels.api as sm

x = sm.add_constant(x)
model = sm.OLS(y, x)
result1 = model.fit()
result1.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.488
Model:                            OLS   Adj. R-squared:                  0.488
Method:                 Least Squares   F-statistic:                     991.0
Date:                Thu, 21 Jul 2022   Prob (F-statistic):               0.00
Time:                        15:20:07   Log-Likelihood:                 19952.
No. Observations:                5195   AIC:                        -3.989e+04
Df Residuals:                    5189   BIC:                        -3.985e+04
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const      -1.819e-05   7.22e-05     -0.252      0.801      -0.000       0.000
x1             0.6612      0.014     47.792      0.000       0.634       0.688
x2             0.0356      0.011      3.291      0.001       0.014       0.057
x3             0.0296      0.016      1.886      0.059      -0.001       0.060
x4             0.0960      0.020      4.779      0.000       0.057       0.135
x5             0.0331      0.009      3.808      0.000       0.016       0.050
==============================================================================
Omnibus:                     1508.823   Durbin-Watson:                   2.816
Prob(Omnibus):                  0.000   Jarque-Bera (JB):           204719.252
Skew:                           0.093   Prob(JB):                         0.00
Kurtosis:                      33.753   Cond. No.                         309.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

We see that the Chinese RMB (against the Franc) closely following the US dollar against the Franc. If the dollar depreciates against the Franc by one percent, the RMB depreciates by .68 percent. 

In [11]:
ehat = result1.resid
SSE = ehat.T.dot(ehat)
SSE

0.14036047829272022

We see that the constant term is insignificant.  The coefficient for the US dollar is significant and large.  The largest weight on a currency after the US dollar is on the Euro, with a weight of .09.  The weights on the Yen, the GBP, and the KRWon are small but significant.  The overall R-squared is .488 and the corrected R-squared is just slightly lower.  The overeall sum of squared residuals is 0.1404.

Now we can test the joint significance of the variables other than the dollar for the Chinese exchange-rate management of the RMB against the Swiss Franc.




$\underline{T-Statistics}$

To test a hypothesis about an estimated coefficient, $\hat{\beta}_{k}$,  we construct the corresponding t-statistic, based on the deviation of the random variable $\hat{\beta}_{k}$  from its (hypothesized) true mean, $\beta_{k0}$, divided by its standard deviation, $\hat{\sigma}_{\hat{\beta}_{k}}$:

$t_{\hat{\beta}_{k}} = \frac{\hat{\beta}_{k} - \beta_{k}}{\hat{\sigma}_{\hat{\beta}_{k}}}$



The standard deviation $\hat{\sigma}_{\hat{\beta}_{k}}$ is simply the square root of the variance of $\hat{\beta}_{k}$:

$\hat{\sigma}_{\hat{\beta}_{k}} = \hat{\sigma} *(x_{k}'x_{k})^{-1}$;

The probability distribution of the t-statistic is function of the degrees of freedom of regression equation as well as the computed t-value, and makes use of the Gamma function, $\Gamma$.  

Letting $\nu = N-k$, the probability distribution of the t-statistic has the following functional form:

$\textbf{Pr}(t_{\hat{\beta}}) = \frac{\Gamma[.5(\nu+1)]}{\sqrt{\nu\pi}\Gamma(.5\nu)} (1+\frac{t_{\hat{beta}}^{2}}{\nu})^{-.5(\nu+1)} $


The probability distribution quickly converges to that of a normal distribution as the degrees of freedom increase. 

The Gamma function is an interpolation for positive real numbers of the factorial function for integers.  Thus for a given positive integer v,

$\Gamma(v+1) = v!$



$\underline{Hypothesis\,Testing\,on\,Parameters}$


In classical statistical decision making, for a given hypothesis, $H_{0} = \beta_{k} = \beta_{k0}$, there are four decisions, two correct ones and two incorrect ones.  The two correct ones are to accept the null if it is true in the real world, and reject it if it is false.  Rejecting the null when it is false is called a Type I error, the probability denoted by $\boldsymbol\alpha$, and accepting it when it is false is call a Type II error, it probability denoted by $\boldsymbol\beta$.


Once we have the estimate of the coefficient $\hat\beta_{k}$ and its standard deviation, $\hat{\sigma}_{\hat{\beta}_{k}}$, it is easy to use the T-distribution to test a hypothesis about the underlying true $\beta_{k}$.  

The hypothesis test proceeds in the following steps for $\textbf{$\alpha$}$, the Probability of a Type I error,  usually set at .05:

1.  Formulate null hypothesis,  $H_(0):\beta_{k} = \beta_{k0}$
2.  Estimate the coefficient $\hat{\beta}_{k}$ and its standard deviation, $\hat{\sigma}_{\hat{\beta}_{k}}$ with OLS.
3.  Set a confidence interval for the cumulative symmetric t-distribution.  Usually we set a two-sided confidence interval of 95%, or a 5% probability of a Type I error, which means selected values for the cumulative t-distribution of [.025, .975] for the degrees of freedom, $\nu$.  
The values we obtain we call the $\textbf{critical values}$ of the t-distribution,  $[t^{*}_{\nu,.025}, t^{*}_{\nu,.975}]$.
4.  Construct a confidence interval for the null hypothesis,  $[\beta_{k0}-t^{*}_{\nu,.025}\hat{\sigma}\_{\hat{\beta}_{k}}, \beta_{k0}+t^{*}_{\nu,.975}\hat{\sigma}_{\hat{\beta}_{k}}]$.  The numbers [.025, .975] represent the t values for $[.5 \cdot \boldsymbol\alpha,  1-.5 \cdot \boldsymbol\alpha]$, respectively.  
5.  Decision:  reject null hypothesis $H_{0}$ if $\hat{\beta}_{k}$ is outside the confidence interval, otherwise fail to reject.  Note, we never say we $\textit{accept}$  the null hypothesis, $H_{0}$, we simply say we fail to reject it. 

Note that the T-tests are tests on each element of the estimated parameter vector, $\hat{\beta_{k}}$.  If there is multicollinearity in any significant degree, thought not perfect multicollinearity, these T-statistics can be misleading, since they do not pick up the effect of other variables other that $x_{k}$ on the standard deviation of $\hat{\beta}$.  


$\underline{Regression\,Diagnostics:Goodness\,of\,Fit\,and\,Overfitting}$



In addition to the point estimates and the t-statistics on each coefficient, we obtain from the regression a common measure of "goodness of fit" called the $R^{2}$ statistic.

$R^{2} =  \frac{Var(\hat{y})}{Var(y)} = 1 -  \frac{Var(\hat\epsilon)}{Var(y)}$

As see above, this statistic can be interpreted either as the ratio of variance of the explained or predicted dependent variable to the variance of actual observed dependent variable, or as one less the ratio of the variance of the residuals to the variance of the actual dependent variable.  

It is a variance-ratio statistic, the measure of the percentage of the total variance explained by the model to the total variance of the dependent variable.  For this reason, $0\leq{R^{2}}\leq{1}$. 

Note however, that this statistic does not make sense if there is no constant term or intercept. Why is this so?  The reason is that an intercept forces both the actual dependent variable, y, and the explained dependent variable, $\hat{y}$, to have the same mean, with $\bar{y} = \bar{\hat{y}}$.  If there is no intercept, the means are different, so that $R^{2}$ statistic is comparing variances centered around different means.  So one is likely to get statistics which are outside the bounds [0,1].

While this statistic is a useful measure of overall goodness of fit, there are limitations and dangers, especially in the multivariate case.  After all, one can always increase the  $R^{2}$ measure by throwing in more regressors.  Why not?

The is the classic problem of over-fitting. The basic message is that models that do exceptionally well with in-sample data (data used to estimate the coefficients), often fail miserably when we use the model for forecasting the dependent variable, with data not used in the estimation.

To make a trivial point, if we have N observations and we use N regressors, we will always get a perfect $\textit{in-sample}$ fit, with $R^{2} = 1$. No big deal.  But if we feed in new regressors, for the given estimated vector, $\hat{\beta}$, such models will generally fail abysmally for $\textit{out-of-sample}$ performance.  We would like good estimated models to do well, both in-sample as well as well out-of-sample.

For this reason we take up the issue of regularization criteria. 



$\underline{Regularization Criteria}$

The basic message is that one can always add in more regressors, and thus more parameters to estimate, to get a better fit.  But in good data science, we look for parsimony, we want to favor simpler models over more complex models, with more and more parameters.  Bigger is not always better.

So how do we compare models with different number of regressors?  Obviously, the goodness of fit measure $R^{2}$ does not take into account the number of regressors being used.  

One measure is the corrected or adjusted R-squared, often denoted as R-bar-squared:

$\bar{R}^{2} = 1-(1-R^{2})\cdot\frac{N-1}{N-k}$

where N is the number of observations and k is the number of regressors.

The adjusted $R^{2}$ statistic is a good start for comparing two models with different numbers of regressors.  Choose the one, not with the better $R^{2}$ but the better $\bar{R}^{2}$.

However, the adjusted $R^{2}$ is not the only statistic for comparing models with different numbers of coefficients or regressors.  Three statistics which often are part of regression output are the Akaike Information Criterion (AIC), the Schwartz Information Criterion (BIC) and the Hannan-Quinn Information Criterion (HQIF).  There statistics are functions of the in-sample sum of squared errors:
$SSE = \hat{\epsilon}'\hat{\epsilon}$

as well as the number of parameters, k and the number of observations of the data used in the regression:



$AIC = \ln(\frac{SSE}{N}) + \frac{2k}{N}  \\
BIC = \ln(\frac{SSE}{N}) + k\frac{\ln(N)}{N} \\
HQIC = \ln(\frac{SSE}{N}) + k\frac{\ln[\ln(N)]}{N}$


The point is to choose the model with the lowest values based on the respective criteria.  The three differ by their penalty factors for the number of parameters, k.  The AIC punishes a high number of factors by the value $\frac{2}{N}$, which many consider too low a penalty factor for using more parameters.  The BIC uses a factor $\frac{\ln(N)}{N}$, considered too harsh a penalty factor.  The HQIC falls between the two penalty factors, with a penalty weight on k equal to $\frac{\ln[\ln(N)]}{N}$.


If one has several models with different numbers of regressors or independent variables, go for the one with the lowest value.  Of course, the ranking of the models may different among the AIC, BIC and HQIF regularization criteria.  Which one should be used?  My preference, and that of many in the world of high-frequency financial econometrics, would be the HQIC criteria.


$\underline{F-Statistics}$

The F-statistic, developed by Sir Rodney Fisher, is an extremely useful statistic in all matters.
Here we discuss two uses:  joint hypothesis tests, and testing for structural change.

$\textit{Joint Hypothesis Tests}$

As we noted above, the T-statistics may be distorted if the regressors in the x-matrix are not perfectly statistically independent, since the matrix (x'x) is not diagonal when there is some degree of multicollinearity.

In many cases with multivariate regression, we find that individual t-statistics will show that a set of variables, individually, are not significant, but taken together, are jointly significant.  

One way of testing the $\textit{joint}$ significance of a subset of regressors, by imposing r zero restrictions on a subset of parameter, is to do the following set of unrestricted and restricted regressions:

1. Run an unrestricted regression, for all k regressors, and obtain the sum of squared errors of the base unrestricted regression, $SSE_{u} = \hat{\epsilon_{u}}'\hat{\epsilon_{u}}$ from the residuals of the regression.  This regressions has (N-k) degrees of freedom.  
2.  One then does a restricted regression, eliminating r regressors or x-variables,  so that the matrix is now of dimension (Nx$k^{*}$) with $k^{*}$<k, and $k^{*}+r =k$, for r restrictions on k parameters  Then obtain the sum of squared residuals from this restricted regressions,
$SSE_{r} = \hat{\epsilon_{r}}'\hat{\epsilon_{r}}$.
3. Then form the ratio for the F-test of r restrictions, 
$ F = \frac{[SSE_{r}-SSE_{u}]/r}{SSE_{u}/N-k}$
4. Compute the critical F-statistic, for the 5 percent level of significance, $F^{r}_{N-k}$ 
5. Reject the null hypothesis of the joint zero restrictions on the r-dimensional subset of coefficients, if the calculate F-statistic, based on the sum of squared residuals of the restricted and unrestricted regressions, is greater than the critical F-statistic under null hypothesis of the zero restrictions.  

This F- statistic is extremely useful.  Variables individually may appear to be insignificant but taken together, with their interaction effects, may be very significant.  

We will illustrate the use of the F-statistic on our Chinese exchange rate example.  

$\textit{Structural Change}$

Another key use of the F-statistic is to test for structural change.  This test is due to Gregory Chow and is called the Chow Test.  It is also known as a test for equality of coefficients across two samples.

In our example, below, we know that the People's Bank of China moved from a US dollar based fixed-exchange rate regime to basket indexing, or setting their exchange rate to a basket of several currencies  Of course, what people and politicians and policy makers say they do is often what they do in practice.  Put another way, there may be official policy change, but little change in underlying economic behavior. 

If we know the timing of a particular change, we can split up a sample of size N into two smaller samples, $N_{1}$ and $N{2}$, with $N_{1}+N_{2}=N$.   We then do three regressions, one with the full sample, with all N observations, and regressions for the first and second smaller samples, $N_{1}$ and $N_{2}$ and calculate the sum of squared errors, SSE, for the three regressions.  Let $SSE_{f}$, $SSE_{1}$, and $SSE_{2}$ denote the sum of squared residuals for the full sample, and the first and second subsets.  The Chow Test has the following expression:

$\textbf{Chow} = \frac{[SSE_{f}-(SSE_{1}+SSE_{2})]/k}{[SSE_{1}+SSE_{2}]/(N_{1}+N_{2}-2k)}$

The Chow statistic under the null hypothesis of equality of coefficients across sub-periods, or no structural change, has an F-distribution with k,N-2k degrees of freedom, since $N_{1}+N_{2}=N$.

$\textbf{Chow} \sim F^{k}_{N-2k}$
  

In time series models, we also need to test for serial independence in the residuals.  In cross-section analysis we test for heteroskedasticity.   We will discuss the various tests in the next two chapters, when we discussion multivariate regression with time series and with cross-section data.

The importance of having residuals as identically and independently distributed random variables cannot be overstated.  If there are detectable non-random patterns in the residuals, this means that there is significant misspecification error in the model.  Put another way, it means that we have omitted key variables.  We would like the residuals to represent "white noise" processes.  

In [10]:
xres = x[:,0:2]
model1 = sm.OLS(y,xres)
result1_restricted = model1.fit()
result1_restricted.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.479
Model:                            OLS   Adj. R-squared:                  0.479
Method:                 Least Squares   F-statistic:                     4777.
Date:                Thu, 21 Jul 2022   Prob (F-statistic):               0.00
Time:                        15:23:11   Log-Likelihood:                 19905.
No. Observations:                5195   AIC:                        -3.981e+04
Df Residuals:                    5193   BIC:                        -3.979e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const      -7.773e-06   7.28e-05     -0.107      0.915      -0.000       0.000
x1             0.7332      0.011     69.116      0.000       0.712       0.754
==============================================================================
Omnibus:                     1509.936   Durbin-Watson:                   2.788
Prob(Omnibus):                  0.000   Jarque-Bera (JB):           194531.170
Skew:                           0.159   Prob(JB):                         0.00
Kurtosis:                      32.977   Cond. No.                         146.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [11]:
resid1 = result1_restricted.resid
SSEres = resid1.T.dot(resid1)
SSEres

0.14292118485270894

We see that the restricted sum of squared errors  is .1429 while the unrestricted sum of squared errors is .1404.  So is there a significant difference?  Is it ok to remove the additional four exchange rates, that of the Yen, GDP, Euro and Korean Won? We can test this with the F-test.

In [12]:
k = 4;
dgf = 5195-6;
Fstat_num =  (.1429-.1404)/k;
Fstat_den  =  .1404/dgf;
Fstat = Fstat_num/Fstat_den
Fstat

23.099180911680932

Is this F-statistic significant?  Can we reject the null that the four eliminated variables do not make any difference? We can do so with the help of the F-distribution for (4,5190) degrees of freedom
Type f.cdf? to find the function

In [13]:
from scipy.stats import f


In [14]:
1-f.cdf(23.1036,6,5190)

1.1102230246251565e-16

Another way is to find the critical F-value for Type I error = .05 for the degrees of freedom, (6, 5190)

In [26]:
f.ppf(.95,k,dgf)

2.3736442966119218

We see that the calculated F statistic is well above the critical F value so we can reject the null hypothesis that the four exchange rates are jointly insignificant.  They matter, for sure.  Maybe not as much as the dollar but they are playing a role in the way the Chinese manage their currency against the Swiss Franc, at least.

Now lets see when we add in the other exchange rates, the Malaysian Ringgit, the Singapore dollar and the Thai Baht.  These currencies, like the RMB, also follow the dollar so they are not independent of each other.  We are asking for trouble when we use these as regressors.  Lets do it anyway to see what happens.  

In [27]:
y = np.diff(np.log(CNYCHF1)) 
x = np.diff(np.log([USDCHF1, YENCHF1, GBPCHF1, EURCHF1, KRWCHF1, MYRCHF1, SGDCHF1, THBCHF1])).T
xx = sm.add_constant(x)

In [28]:
model2 = sm.OLS(y,xx);
result2 = model2.fit()
result2.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.539
Model:                            OLS   Adj. R-squared:                  0.538
Method:                 Least Squares   F-statistic:                     757.5
Date:                Tue, 15 Feb 2022   Prob (F-statistic):               0.00
Time:                        19:32:37   Log-Likelihood:                 20221.
No. Observations:                5195   AIC:                        -4.042e+04
Df Residuals:                    5186   BIC:                        -4.036e+04
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const      -3.815e-06   6.86e-05     -0.056      0.956      -0.000       0.000
x1             0.4886      0.015     32.377      0.000       0.459       0.518
x2             0.0011      0.011      0.098      0.922      -0.020       0.022
x3            -0.0069      0.015     -0.462      0.644      -0.036       0.022
x4            -0.0006      0.020     -0.030      0.976      -0.039       0.038
x5             0.0140      0.009      1.558      0.119      -0.004       0.032
x6            -0.0088      0.002     -3.755      0.000      -0.013      -0.004
x7             0.3201      0.014     22.293      0.000       0.292       0.348
x8             0.1016      0.012      8.657      0.000       0.079       0.125
==============================================================================
Omnibus:                     1609.554   Durbin-Watson:                   2.794
Prob(Omnibus):                  0.000   Jarque-Bera (JB):           302842.944
Skew:                           0.106   Prob(JB):                         0.00
Kurtosis:                      40.404   Cond. No.                         318.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Now we see something very strange. The Chinese weight on the US dollar has dropped, they are now following the Singapore Dollar with a weight of .32 and the Thai baht with a weight of .10. Can this be true?  Do you really think that the People's Bank of China pays almost the same attention to the Sing dollar as they do to the US dollar?  
This is a care when we have to bring $\textit{priors}$ to our econometric analysis.   We have to make use of prior knowledge that these currencies are not independent of each other and the US dollar,  and thus should not be used as independent regressors seeking to explain the behavior of RMB.  

Now what about structural change?  Did the coefficients change, using the original set of coefficients, before and after the period of Aug. 1, 2005?  Date 1516 marks Aug. 1, 2005 at the index of 1516 (see cell below)  So we will do three regressions and calculate the sum of squared and do a Chow test. 

In [29]:
date2[1515]

numpy.datetime64('2005-08-01T00:00:00.000000000')

In [30]:
y = np.diff(np.log(CNYCHF1))
x = np.diff(np.log([USDCHF1, YENCHF1, GBPCHF1, EURCHF1, KRWCHF1])).T
x = sm.add_constant(x)

results_all = sm.OLS(y,x).fit()
results_before = sm.OLS(y[:1514],x[:1514,:]).fit()
results_after = sm.OLS(y[1515:],x[1515:,:]).fit()
error_all = results_all.resid
error_before = results_before.resid
error_after = results_after.resid
SSE_all = error_all.T.dot(error_all)
SSE_before = error_before.T.dot(error_before)
SSE_after = error_after.T.dot(error_after)

In [31]:
[SSE_all, SSE_before, SSE_after]

[0.14036047829272022, 0.0357119299288346, 0.09895188016321668]

We see that there is a big jump in the errors, when we used the combined full sample.  This indicates that there is some type of underlying structural break.  We can use the Chow test, with k = 12 degrees of freedom for the numerator and 5196-12 degrees of freedom for the denominator.

In [33]:
ChowNum =  (SSE_all-(SSE_before+SSE_after))/6;
ChowDen =   (SSE_before+SSE_after)/(5196-12);
Chow = ChowNum/ChowDen;
Chow

36.549696032018694

This Chow test is pretty high and we can guess that it is significant, that we can reject the hypothesis of no structural change.  Lets calculate the critical F value under the null hypothesis.

In [34]:
f.ppf(.95, 6, 5196-12)

2.1003375837818896

We see that the calculated F statistic is far above the critical value under the null hypothesis.So something significant did happen at the PBC (People's Bank of China) exchange-rate department after August. 2005.  

$\textbf{Summary}$

We explored how F-statistics can be used in the context of multivariate regression for testing the joint significance of several variables at a time as well as for testing for a structural break.
We did so in the context of time-series regression.

The test for a structural break can also be put to good use in a cross-section context. For example, for uncovering discrimination in labor markets across industries, one can sample the wage changes of men and women, with regressors for age, experience, education, region of the country, macroeconomic conditions.  One way to uncover evidence of discrimination is to test equality of coefficients between the genders, to see if there is different processes at work for wage changes.  

We did not discuss dummy variables for examining structural change.  We can do this by simply putting in a regressor, having the value 0 before Aug. 2005 and 1 afterwards.  But the Chow test is more flexible and general.  

As an exercise, I would suggest you do the same exercise but use the AUD (Australian Dollar) as the numeraire currency, rather than the Swiss Franc (CHF).   The PBC did say that neither the Aus dollar nor the Swiss franc were in their basket for currency indexing.


